lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -115,6 +116,15 @@ class SideEffectsFlagPlugin {
 								if (mapping) {
 									dep.redirectedModule = mapping.module;
 									dep.redirectedId = mapping.exportName;
+									module.reasons.splice(i--, 1);
```

Comment:
This is really bad for perfs. Would it be possible to build a second array or to modify `module.reasons` in a second loop?

Reply:
I'll do this in a separate PR. I want to fix the bug first.